### PR TITLE
Predict type set get

### PR DESCRIPTION
**Pull request by mb706**

### CODE REVIEW

File: R/GraphLearner.R [hunk 1]
```diff
@@ -102,6 +94,37 @@ GraphLearner = R6Class("GraphLearner", inherit = Learner,
...
+    },
+    get_predict_type = function() {
+      # recursively walk backwards through the graph 
+      get_po_predict_type = function(x) {
+        if (!is.null(x$predict_type)) return(x$predict_type)
+        prdcssrs = self$graph$edges[dst_id == x$id, ]$src_id
+        if (length(prdcssrs)) {
+          # all non-null elements
+          predict_types = keep(map(self$graph$pipeops[prdcssrs], get_po_predict_type), Negate(is.null))
+          if (length(unique(predict_types)) == 1L)
```
Comment (mb706) [inline R/GraphLearner.R:106]: its a bit weird to ignore NULL-returns by `get_po_predict_type` calls, but *also* encode inconsistent predecessor predict types as `NULL`.

File: R/GraphLearner.R [hunk 2]
```diff
@@ -102,6 +94,37 @@ GraphLearner = R6Class("GraphLearner", inherit = Learner,
         return(value$clone(deep = TRUE))
       }
       value
+    },
+    get_predict_type = function() {
+      # recursively walk backwards through the graph 
+      get_po_predict_type = function(x) {
+        if (!is.null(x$predict_type)) return(x$predict_type)
```
Comment (mb706) [inline R/GraphLearner.R:101]: `x$predict_type` should go into the base class

File: R/GraphLearner.R [hunk 3]
```diff
@@ -102,6 +94,37 @@ GraphLearner = R6Class("GraphLearner", inherit = Learner,
         return(value$clone(deep = TRUE))
       }
       value
+    },
+    get_predict_type = function() {
```
Comment (mb706) [inline R/GraphLearner.R:98]: comment smth like "trace every path backward from the sink until a `x$predict_type` is found for each of these paths (or there are no predecessors)"

File: R/PipeOpLearner.R [hunk 4]
```diff
@@ -118,6 +119,13 @@ PipeOpLearner = R6Class("PipeOpLearner", inherit = PipeOp,
         }
       }
       private$.learner
+    },
+    predict_type = function(val) {
+      if (!missing(val)) {
+        assert_subset(val, names(mlr_reflections$learner_predict_types[[private$.learner$task_type]]))
```
Comment (mb706) [inline R/PipeOpLearner.R:125]: doesn't the learner itself also assert this?

Comment (pfistfl) [reply R/PipeOpLearner.R:125]: pipeops can / could have AB's that are not learner's. But those might assert this themselves aswell.

Comment (pfistfl) [reply R/PipeOpLearner.R:125]: we should think about what PipeOpThreshold does in this case. There we can not set to `prob`.
this should then possibly error (within PipeOpThreshold I guess).

File: R/GraphLearner.R (hunk 1 above)
Comment (pfistfl) [reply R/GraphLearner.R:106]: true

**Comment by mb706** [comment]
closes #306

**Comment by mb706** [comment]
broken because not checked out from master

**Comment by mb706** [comment]
(superseded by #457)

**Pull request closed without merging**